Ready

justnapa opened the pull request:
все работает

justnapa committed:
app run
(03/07 20:13)

justnapa committed:
app run
(03/07 20:13)

justnapa committed:
with error (04/07 morning)

estetika-panelek commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/filter/model/FilterContactTypeUi.java:
```diff
@@ -20,6 +22,17 @@ public boolean isSelected() {
         return selected;
     }
 
+
+
+
+
```
🍏  Излишние импорты лучше не делать. Кроме того, можешь применить в студии автоформатирование (в выбранном файле, сверху вкладка Code - Reformat Code, либо Ctrl+Alt+L), тогда автоматически код выправится по стандартм форматирования

justnapa replied:
Извиняюсь за топорное исполнение - времени было мало, я еще отстал за отпуск. На каком то моменте что-то пошло не так с Git, я пытался сделать новые ветки, откатил до начала, зафиксировал, потом вернулся сделал PR , я, очевидно, что-то недопонимаю. Выслал Вам папку с проектом - у меня все работает без ошибок, примите работу, пожалуйста

estetika-panelek commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/main/ContactAdapter.java:
```diff
@@ -23,12 +23,13 @@
     private final AsyncListDiffer<ContactUi> differ = new AsyncListDiffer<>(
             new AdapterListUpdateCallback(this),
-            new AsyncDifferConfig.Builder<>(new ListDiffCallback()).build()
+            new AsyncDifferConfig.Builder<>(new BaseListDiffCallBack<ContactUi>()).build()
```
🍏 можешь удалять ListDiffCallback, который лежит в этом файле, и остальные ListDiffCallback, тк они не используются

justnapa replied:
у меня нет этих строк с ListDiffCallBack
![2024-07-07_19-40-01](https://github.com/justnapa/android-contacts-interfaces-lab/assets/38713041/52006b04-d78c-44ef-9d5b-618fda1bf2e8)

estetika-panelek replied:
ниже в файле в 97 строке. Можешь воспользоваться поиском по тексту в Android Studio

estetika-panelek commented on app/src/main/java/ru/yandex/practicum/contacts/SplashActivity.java:
```diff
@@ -33,13 +34,13 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         setContentView(binding.getRoot());
-        binding.settingsButton.setOnClickListener(view -> navigateToSettings());
-
-        if (ContextUtils.hasContactPermissions(this)) {
-            navigateToMain();
-        } else {
-            requestPermissionLauncher.launch(Manifest.permission.READ_CONTACTS);
-        }
```
⚠️ Был удален код навигации, теперь твое приложение не открывается, открывается только слеш.

estetika-panelek replied:
🍏 Приложение открывается, но лучше было бы вернуть удаленный из onCreate код, а не добавлять код в onStart. Ты можешь на гитхабе во вкладке Files changed посмотреть, какой код ты удалял, а какой добавлял

estetika-panelek commented:
🍏  На будущее, лучше называть коммиты осмысленно, писать в них, что конкретно было сделано

justnapa committed:
with error (04/07 morning)

justnapa committed:
добавил в SplashActivity
else {
            requestPermissionLauncher.launch(Manifest.permission.READ_CONTACTS);
        } 07/07/24
